cpp/src/arrow/util/thread_pool.h

```diff
@@ -189,6 +190,64 @@ class ARROW_EXPORT Executor {
                            StopCallback&&) = 0;
 };
 
+/// \brief An executor implementation that runs all tasks on a single thread using an
+/// event loop.
+///
+/// Note: Any sort of nested parallelism will deadlock this executor.  Blocking waits are
```

Comment:
This is true of a parallel executor as well, no?

Reply:
True, I added the comment to the other thread pool as well.